t/plugin/ai3.t

```diff
@@ -0,0 +1,282 @@
+deployment:
+  role: traditional
+  role_traditional:
+    config_provider: etcd
+  admin:
+    admin_key: null
+apisix:
+  node_listen: 1984
+plugins:
+            ]]
+            require("lib.test_admin").set_config_yaml(data)
+
+            return t('/apisix/admin/plugins/reload',
+                                    ngx.HTTP_PUT)
+        end
+
+        load_ai_module = function ()
+            local t = require("lib.test_admin").test
+            local data = [[
+deployment:
+  role: traditional
+  role_traditional:
+    config_provider: etcd
+  admin:
+    admin_key: null
+apisix:
+  node_listen: 1984
+plugins:
+  - ai
+            ]]
+            require("lib.test_admin").set_config_yaml(data)
+
+            return t('/apisix/admin/plugins/reload',
+                                        ngx.HTTP_PUT)
+        end
+_EOC_
+
+        $block->set_value("extra_init_by_lua", $extra_init_by_lua);
+    }
+
+    if ((!defined $block->error_log) && (!defined $block->no_error_log)) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: enable(default) -> disable -> enable
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua_block {
+            local ai = require("apisix.plugins.ai")
+            local router = require("apisix.router")
+            local org_match = router.router_http.matching
+            local ai_match = ai.route_matching
+
+            local apisix = require("apisix")
+            local org_upstream = apisix.handle_upstream
+            local ai_upstream = ai.handle_upstream
+
+            local org_balancer_phase = apisix.http_balancer_phase
+            local ai_balancer_phase = ai.http_balancer_phase
+
+            local t = require("lib.test_admin").test
+            -- register route
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "host": "127.0.0.1",
+                    "methods": ["GET"],
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uri": "/hello"
+                }]]
+            )
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            local code = t('/hello', ngx.HTTP_GET)
+            assert(code == 200, "enable(default): access /hello")
+            assert(router.router_http.matching == ai_match, "enable(default): router_http.matching")
+            assert(apisix.handle_upstream == ai_upstream, "enable(default): ai_upstream")
+            assert(apisix.http_balancer_phase == ai_balancer_phase, "enable(default): http_balancer_phase")
```

Comment:
it is not a good style, you are writing E2E test case
we can not use `unit test case` way